rpcserver.go

```diff
@@ -720,24 +720,21 @@ func (r *rpcServer) OpenChannel(in *lnrpc.OpenChannelRequest,
 
 	// TODO(roasbeef): also return channel ID?
 
-	// If the node key is set, then we'll parse the raw bytes into a pubkey
-	// object so we can easily manipulate it. If this isn't set, then we
-	// expected the TargetPeerId to be set accordingly.
-	if len(in.NodePubkey) != 0 {
-		nodePubKey, err = btcec.ParsePubKey(in.NodePubkey, btcec.S256())
-		if err != nil {
-			return err
-		}
-
-		// Making a channel to ourselves wouldn't be of any use, so we
-		// explicitly disallow them.
-		if nodePubKey.IsEqual(r.server.identityPriv.PubKey()) {
-			return fmt.Errorf("cannot open channel to self")
-		}
+	// Parse the raw bytes of the node key into a pubkey object so we
+	// can easily manipulate it.
+	nodePubKey, err = btcec.ParsePubKey(in.NodePubkey, btcec.S256())
```

Comment:
In case the key is not set by the caller (is empty), we could return an error like "NodePubKey not set". I guess the parsing will fail in any case, but not sure how intelligible the returned error will be for the caller.